serverlessworkflow/modules/ROOT/pages/use-cases/orchestration-based-saga-pattern.adoc

```diff
@@ -1 +1,269 @@
-//= Orchestration-based SAGA pattern
\ No newline at end of file
+= Orchestration-based SAGA pattern
+=====================================================
+06-27-2022
+:compat-mode!:
+// Metadata:
+:description: Saga Pattern use case example
+:keywords: kogito, workflow, serverless, saga, saga pattern, consistency, microservices, orchestration, orchestrator
+
+== Brief introduction to Saga pattern
+
+Saga is a design pattern to manage data consistency across participants in distributed transaction scenarios.
+
+In a microservice architecture, a participant could be defined as a microservice, responsible to perform actions related to its business domain.
+Saga is meant to provide transaction management using a sequence of steps.
+
+While executing a step, failures might happen, in this situation, a sequence of compensating actions are executed to undo the changes that were made during the execution, or to leave the system in a known termination state, aiming for consistency.
+
+== The order fulfillment example
+
+This is a very common use case, present in many e-commerce companies and for sure it should be familiar to you.
+The basic idea in this scenario is a user buys some item through an e-commerce application inputting the payment method and delivery information and waits for this item to be delivered at the specified address.
+Given this use case, what happens behind the scenes is a sequence of steps that should be executed to complete the order, as shown in the picture:
+
+.Example Order fulfillment
+image::use-cases/orchestration-based-saga-pattern/order-fulfillment-example.png[]
+
+In the example there are mainly the following steps:
+
+<1> _Order checkout_: the user confirms the order with the chosen items, payment method, and delivery address.
+<2> _Stock reservation_: where this item from the order is selected from the stock and reserved, being removed from the total stock item.
+<3> _Payment processing_: based on the payment method and the user information, for instance, the credit card, the credit is checked and allocated to that payment.
+<4> _Shipping processing_: it represents a mechanism that could communicate with third-party systems responsible to deliver orders or even an internal system responsible for logistics and delivery.
+
+In the below picture, it is possible to visualize the two main flows in case of success or failure, during the order fulfillment flow which describes a Saga pattern.
```

Comment:
```suggestion
The below picture makes it possible to visualize the two main flows in case of success or failure during the order fulfillment flow, which describes a Saga pattern.
```